### DISCUSSION TITLE
Show the location of the recognize word in the original image

### DISCUSSION THREAD

**Discussion by khawar-islam** [mindee/doctr#1071] [General]
How we can recognize complete word and write down on the top of the word? I directly want to save the image after recognize, I do not want to look through matplotlib.

![image](https://user-images.githubusercontent.com/11488932/191415674-a4a3a3a4-b910-4ea5-89f6-0a193d22c338.png)

**Comment by felixdittrich92** [comment]
Hi @khawar-islam :wave:,
About the matplotlib display issue for korean : https://stackoverflow.com/questions/48374379/display-foreign-language-in-matplotlib-in-virtual-env
Otherwise you can write your custom code and perform the same with PIL as described in the other question with using the pipeline you get all you need : box coords / words / confidence :) simply use `out.export() `

**Reply by khawar-islam** [reply]
Hello @felixdittrich92 finally, i fixed the `matplotlib` error and now i can see korean characters. I trained a model on `vocab = "".join(set(VOCABS['english'] + VOCABS['korean']))` vocabs for 5 epochs and when i tested image English characters becomes Korean and recognized korean characters are wrong. **Is it because for 5 epochs?**

![Screenshot from 2022-09-22 09-46-23](https://user-images.githubusercontent.com/11488932/191640621-4028ffdb-b64d-4df8-af18-7d4837e73c75.png)

**Reply by felixdittrich92** [reply]
Hi @khawar-islam does your dataset contain both english and korean samples ? About the wrong recognized chars i think you need a more real dataset. Have you used your synthetic generated dataset or our WordGenerator ? :)

**Reply by khawar-islam** [reply]
Hi @felixdittrich92  Oh, there is only korean samples available in my dataset. It means that i have to modify and only korean vocabs are given like `vocab = "".join(set(VOCABS['korean']))`. I used my synthetic generated dataset because it is especially design for korean language i found in other repo. I have no idea whether i am wrong or right. 

What is your suggestion? Can i use your WordGenerator? Or increase dataset size or more epochs? If i increase more handwritten fonts during training, is it worth-full?

**Reply by felixdittrich92** [reply]
Hi @khawar-islam yeah sure you need samples in your dataset which contains the chars in images you want to train. For example korean: needs images with korean text, english needs samples with english text, ...

You could take a look at: https://github.com/youngkyung/SynthText_kr (word images) 
Unfortunately, there is no right or wrong ... the more data you have, and the closer you are to real examples, the better the model will perform in the end

**Reply by felixdittrich92** [reply]
The WordGenerator needs some improvements and is only useful for pretraining currently

**Reply by khawar-islam** [reply]
Hello @felixdittrich92, please have a look on complete training and results. What is your opinion?

```
Epoch 1/25 - Validation loss: 0.0451385 (Exact: 96.29% | Partial: 96.29%)
Validation loss decreased 0.0451385 --> 0.0279511: saving state...                                                                       
Epoch 2/25 - Validation loss: 0.0279511 (Exact: 97.82% | Partial: 97.82%)
Validation loss decreased 0.0279511 --> 0.0277946: saving state...                                                                       
Epoch 3/25 - Validation loss: 0.0277946 (Exact: 98.13% | Partial: 98.13%)
Validation loss decreased 0.0277946 --> 0.0249508: saving state...                                                                       
Epoch 4/25 - Validation loss: 0.0249508 (Exact: 98.32% | Partial: 98.32%)
Epoch 5/25 - Validation loss: 0.029055 (Exact: 98.60% | Partial: 98.60%)                                                                 
Epoch 6/25 - Validation loss: 0.0313421 (Exact: 98.74% | Partial: 98.74%)                                                                
Epoch 7/25 - Validation loss: 0.0315991 (Exact: 98.76% | Partial: 98.76%)                                                                
Epoch 8/25 - Validation loss: 0.0280943 (Exact: 98.84% | Partial: 98.84%)                                                                
Epoch 9/25 - Validation loss: 0.0379387 (Exact: 98.84% | Partial: 98.84%)                                                                
Epoch 10/25 - Validation loss: 0.0414532 (Exact: 98.87% | Partial: 98.87%)                                                               
Epoch 11/25 - Validation loss: 0.0285905 (Exact: 98.89% | Partial: 98.89%)                                                               
Epoch 12/25 - Validation loss: 0.0397933 (Exact: 98.91% | Partial: 98.91%)                                                               
Epoch 13/25 - Validation loss: 0.0378084 (Exact: 98.92% | Partial: 98.92%)
Epoch 14/25 - Validation loss: 0.0378055 (Exact: 98.95% | Partial: 98.95%)                                                               
Epoch 15/25 - Validation loss: 0.0328173 (Exact: 98.95% | Partial: 98.95%)                                                                   
Epoch 16/25 - Validation loss: 0.0498214 (Exact: 98.97% | Partial: 98.97%)                                                                   
Epoch 17/25 - Validation loss: 0.0540435 (Exact: 98.98% | Partial: 98.98%)                                                                   
Epoch 18/25 - Validation loss: 0.0396671 (Exact: 98.99% | Partial: 98.99%)                                                                   
Epoch 19/25 - Validation loss: 0.0490498 (Exact: 98.99% | Partial: 98.99%)                                                                   
Epoch 20/25 - Validation loss: 0.0484238 (Exact: 99.00% | Partial: 99.00%)                                                                   
Epoch 21/25 - Validation loss: 0.0531673 (Exact: 99.00% | Partial: 99.00%)                                                                   
Epoch 22/25 - Validation loss: 0.0540416 (Exact: 99.00% | Partial: 99.00%)                                                                   
Epoch 23/25 - Validation loss: 0.0527688 (Exact: 99.01% | Partial: 99.01%)                                                                   
Epoch 24/25 - Validation loss: 0.0527038 (Exact: 99.01% | Partial: 99.01%)                                                                   
Epoch 25/25 - Validation loss: 0.0525234 (Exact: 99.01% | Partial: 99.01%)   
```
I have also test above model on images but unfortunately the results is almost poor. Even on computer text-based the model unable to recognize correctly.
![Screenshot from 2022-09-26 11-31-55](https://user-images.githubusercontent.com/11488932/192182470-4c6aa3d3-84c6-47b1-9a21-c65557553199.png)
